File: app/src/main/java/io/github/coffeegerm/glucoseguide/ui/statistics/children/OneMonthStatisticsFragment.java
/*
 * Copyright 2018 Coffee and Cream Studios
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package io.github.coffeegerm.glucoseguide.ui.statistics.children;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.Calendar;
import java.util.Date;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;
import io.github.coffeegerm.glucoseguide.GlucoseGuide;
import io.github.coffeegerm.glucoseguide.R;
import io.github.coffeegerm.glucoseguide.data.DatabaseManager;
import io.github.coffeegerm.glucoseguide.utils.Utilities;

/**
 * Created by dyarz on 8/15/2017.
 * <p>
 * Fragment for Statistics ViewPager to show
 * one month data for user.
 */

public class OneMonthStatisticsFragment extends Fragment {
  
  @Inject
  public SharedPreferences sharedPreferences;
  @Inject
  public Utilities utilities;
  @Inject
  public DatabaseManager databaseManager;
  
  @BindView(R.id.average)
  TextView average;
  @BindView(R.id.highest)
  TextView highest;
  @BindView(R.id.lowest)
  TextView lowest;
  
  @Nullable
  @Override
  public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
    View oneMonth = inflater.inflate(R.layout.fragment_one_month_statistics, container, false);
    GlucoseGuide.syringe.inject(this);
    ButterKnife.bind(this, oneMonth);
    setValues();
    return oneMonth;
  }
  
  private void setValues() {
    if (databaseManager.getAllFromDate(getOneMonthAgo()).size() == 0) {
      average.setText(R.string.dash);
      highest.setText(R.string.dash);
      lowest.setText(R.string.dash);
    } else {
      average.setText(String.valueOf(databaseManager.getAverageGlucose(getOneMonthAgo())));
      highest.setText(String.valueOf(databaseManager.getHighestGlucose(getOneMonthAgo())));
      lowest.setText(String.valueOf(databaseManager.getLowestGlucose(getOneMonthAgo())));
    }
  }
  
  private Date getOneMonthAgo() {
    Calendar calendar = Calendar.getInstance();
    calendar.add(Calendar.DATE, -30);
    return calendar.getTime();
  }
}

File: app/src/main/java/io/github/coffeegerm/glucoseguide/ui/statistics/children/OneMonthStatisticsFragment.kt
/*
 * Copyright 2018 Coffee and Cream Studios
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package io.github.coffeegerm.glucoseguide.ui.statistics.children

import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import io.github.coffeegerm.glucoseguide.GlucoseGuide
import io.github.coffeegerm.glucoseguide.R
import io.github.coffeegerm.glucoseguide.data.viewModel.StatisticsViewModel
import io.github.coffeegerm.glucoseguide.data.viewModel.StatisticsViewModelFactory
import io.github.coffeegerm.glucoseguide.utils.DateAssistant
import kotlinx.android.synthetic.main.fragment_one_month_statistics.*
import java.util.*
import javax.inject.Inject

class OneMonthStatisticsFragment : Fragment() {

  @Inject
  lateinit var dateAssistant: DateAssistant
  @Inject
  lateinit var statisticsViewModelFactory: StatisticsViewModelFactory
  private lateinit var statisticsViewModel: StatisticsViewModel

  private lateinit var oneMonthAgo: Date

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    GlucoseGuide.syringe.inject(this)
    statisticsViewModel = ViewModelProviders.of(this, statisticsViewModelFactory).get(StatisticsViewModel::class.java)
  }

  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? = inflater.inflate(R.layout.fragment_one_month_statistics, container, false)

  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
    super.onViewCreated(view, savedInstanceState)
    oneMonthAgo = dateAssistant.getOneMonthAgo()
  }

  override fun onResume() {
    super.onResume()
    getStatistics()
  }

  private fun getStatistics() {
    if (statisticsViewModel.getNumberOfEntries(oneMonthAgo) != 0) {
      one_month_average.text = statisticsViewModel.getAverageBloodGlucose(oneMonthAgo)
      one_month_highest.text = statisticsViewModel.getHighestBloodGlucose(oneMonthAgo)
      one_month_lowest.text = statisticsViewModel.getLowestBloodGlucose(oneMonthAgo)
    }
  }


}
